### ISSUE TITLE
Unable to connect with Dart Observatory when doing automation on real iOS device

### ISSUE THREAD

**Issue by ahashmi24** [truongsinh/appium-flutter-driver#39]
## Failing to make a connection with Dart Observatory due to error TCPConnectWrap.afterConnect [as oncomplete] (net.js:1128:14)

**Note:**  I am able to run my test on the iOS simulator by using same setup and on the same environment.

## Environment

Appium verison: 1.16.0
iOS verison:  13.1.2
xcode version: 11.1

## Link to Appium logs
https://gist.github.com/ahashmi24/bf77b21e30ba2e509e421e4442883842

## Using the below capabilities:
capabilities.setCapability(IOSMobileCapabilityType.BUNDLE_ID, "com.example.datePicket");
		   capabilities.setCapability(MobileCapabilityType.AUTOMATION_NAME, "Flutter");
		    capabilities.setCapability(MobileCapabilityType.PLATFORM_NAME, "iOS");
		    capabilities.setCapability(MobileCapabilityType.ORIENTATION, "PORTRAIT");
		   // capabilities.setCapability("connectHardwareKeyboard", false);
		    capabilities.setCapability(MobileCapabilityType.UDID, "*********************");
		    capabilities.setCapability("includeSafariInWebviews", true);
		    capabilities.setCapability("platformVersion", "13.1.2");
		    capabilities.setCapability(MobileCapabilityType.DEVICE_NAME, "iPhone 8 plus");
		    capabilities.setCapability("noReset", true);
		    driver = new IOSDriver<MobileElement>(new URL("http://127.0.0.1:4723/wd/hub"), capabilities);

**Comment by ahashmi24** [comment]
CC: @KazuCocoa

**Comment by ahashmi24** [comment]
@shibupanda Hi
Did you try with an iOS real device?

Can you plz help me here?

**Comment by shibupanda** [comment]
@ahashmi24
Frankly speaking I didn't try on iOS with these combination. On vacation now.

Still I have some query did you try with released version of application or debug application. I am not sure but mostly this type of issue will come when we try on released version of flutter app.

**Comment by vijaydeepak-rezi** [comment]
Similar Issue https://github.com/truongsinh/appium-flutter-driver/issues/18
CC: @hongbiangoal

**Comment by ahashmi24** [comment]
@shibupanda 
I am using a debug version of the app. I am able to do automation on a simulator with the same build.

**Comment by ahashmi24** [comment]
@hongbiangoal
can you please explain to us how you fixed this issue?

Thanks in advance!

**Comment by vijaydeepak-rezi** [comment]
@ahashmi24 
you need to use iproxy localport deviceport to redirect it to Dart Observatory

Example : iproxy  57172 57172  in a new terminal while appium running on other

**Comment by ahashmi24** [comment]
@vijaydeepak-rezi  Thank you!
Can you plz share code snippet on how to handle this? It would be a great help.

Thanks in advance.

**Comment by shibupanda** [comment]
@ahashmi24 
It should be fixed from driver side but the time being please follow @vijaydeepak-rezi comment.
please open different terminal and run iproxy 57172 57172 while running appium . It's port forwarding from local port to device port.

**Comment by ahashmi24** [comment]
@shibupanda @vijaydeepak-rezi Thank you so much.

But it takes a lot of time to "accept connection" .

@shibupanda shall I keep this ticket open until we get the fix in driver?

**Comment by shibupanda** [comment]
@ahashmi24 Let's create a new issue for this and give this link for the reference.

**Comment by pradarttana-sharechat** [comment]
Hello all,

I have followed the instruction. 

iproxy <observatory_port> <observatory_port> <udid>

I am getting 

`waiting for connection
accepted connection, fd = 4
waiting for connection
Number of available devices == 1
Requesting connecion to device handle == 124 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57388
run_ctos_loop: fd = 4
run_stoc_loop: fd = 4
recv failed: Connection reset by peer
recv failed: Operation not permitted
accepted connection, fd = 4
waiting for connection
Number of available devices == 1
Requesting connecion to device handle == 124 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57388
Error connecting to device!`

sometimes I am getting 

`waiting for connection
accepted connection, fd = 4
waiting for connection
Number of available devices == 2
Requesting connecion to device handle == 126 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57841
Error connecting to device!
accepted connection, fd = 4
waiting for connection
Number of available devices == 2
Requesting connecion to device handle == 126 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57841
Error connecting to device!
accepted connection, fd = 4
waiting for connection
Number of available devices == 2
Requesting connecion to device handle == 126 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57841
Error connecting to device!
accepted connection, fd = 4
waiting for connection
Number of available devices == 2
Requesting connecion to device handle == 126 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57841
Error connecting to device!
accepted connection, fd = 4
waiting for connection
Number of available devices == 2
Requesting connecion to device handle == 126 (serial: a674920a60bb4efd14f1c7824dca56609b4e5cd4), port 57841`

don't know what I am doing wrong. 
in the flutter appium log also i am seeing 
`[XCUITest] Skipping setting of the initial display orientation. Set the "orientation" capability to either "LANDSCAPE" or "PORTRAIT", if this is an undesired behavior.
[debug] [BaseDriver] Event 'orientationSet' logged at 1580730414224 (17:16:54 GMT+0530 (India Standard Time))
[FlutterDriver] Attempt #1
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:58423/Hz1F6oL7tcM=/ws
[FlutterDriver] Error: connect ECONNREFUSED 127.0.0.1:58423
[FlutterDriver]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1106:14)
[FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:58423/Hz1F6oL7tcM=/ws
[FlutterDriver] Waiting 300 seconds before trying...`

@ahashmi24 @shibupanda please help here.

**Comment by ahashmi24** [comment]
@pradarttana-sharechat 
you have to wait for 300 seconds or 5 mints to complete Attempt 1 if you are following the below steps:

1. Get an observatory port from appium server logs.
2. Open a new terminal and hit iproxy [boservatory_port] [boservatory_port] command.
3. Wait for 300 seconds to hit Attempt 2.

**Comment by pradarttana-sharechat** [comment]
@ahashmi24 
in the second attempt. the connection is successful. then again the session is deleted

`[FlutterDriver] Attempt #1
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:55130/8ecI9cXWD1M=/ws
[FlutterDriver] Error: connect ECONNREFUSED 127.0.0.1:55130
[FlutterDriver]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1106:14)
[FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:55130/8ecI9cXWD1M=/ws
[FlutterDriver] Waiting 300 seconds before trying...
[FlutterDriver] Attempt #2
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:55130/8ecI9cXWD1M=/ws
[FlutterDriver] Connected to ws://127.0.0.1:55130/8ecI9cXWD1M=/ws
[Appium] New FlutterDriver session created successfully, session 060453e2-2368-4b53-9da9-d4424232699d added to master session list
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1580799273820 (12:24:33 GMT+0530 (India Standard Time))
[debug] [MJSONWP (060453e2)] Cached the protocol value 'MJSONWP' for the new session 060453e2-2368-4b53-9da9-d4424232699d
[debug] [MJSONWP (060453e2)] Responding to client with driver.createSession() result: {"shell":null,"allowCors":false,"reboot":false,"ipa":null,"address":"0.0.0.0","port":4723,"callbackAddress":null,"callbackPort":null,"bootstrapPort":4724,"backendRetries":3,"sessionOverride":false,"launch":false,"logFile":null,"loglevel":"debug","logTimestamp":false,"localTimezone":false,"logNoColors":false,"webhook":null,"safari":false,"defaultDevice":false,"forceIphone":false,"forceIpad":false,"automationTraceTemplatePath":null,"instrumentsPath":null,"nodeconfig":null,"robotAddress":"0.0.0.0","robotPort":-1,"selendroidPort":8080,"chromeDriverPort":null,"chromedriverExecutable":null,"showConfig":false,"noPermsCheck":false,"enforceStrictCaps":false,"isolateSimDevice":false,"tmpDir":"/var/folders/wg/1m1g5nld5zbb8n7rsqx8t4_c0000gn/T","traceDir":null,"debugLogSpacing":false,"suppressKillServer":false,"longStacktrace":false,"webkitDebugProxyPort":27753,"wdaLocalPort":8100,"defaultCapabilities":{},"relaxedSecurityEnabled":false,"allowInsecure":[],"denyInsecure":[],"defaultCommandTimeout":60,"keepArtifacts":fals...
[HTTP] <-- POST /wd/hub/session 200 318676 ms - 2146
[HTTP] 
Feb 04, 2020 12:24:33 PM io.appium.java_client.remote.AppiumCommandExecutor$1 lambda$0
INFO: Detected dialect: OSS
[HTTP] --> DELETE /wd/hub/session/060453e2-2368-4b53-9da9-d4424232699d
[HTTP] {}
[debug] [MJSONWP (060453e2)] Calling AppiumDriver.deleteSession() with args: ["060453e2-2368-4b53-9da9-d4424232699d"]
[debug] [BaseDriver] Event 'quitSessionRequested' logged at 1580799273928 (12:24:33 GMT+0530 (India Standard Time))
[Appium] Removing session 060453e2-2368-4b53-9da9-d4424232699d from our master session list
[debug] [FlutterDriver] Deleting Flutter Driver session
[debug] [WD Proxy] Matched '/session/e6b30394-6ad7-423c-aad8-f1df82edc2af' to command name 'deleteSession'
[debug] [WD Proxy] Proxying [DELETE /session/e6b30394-6ad7-423c-aad8-f1df82edc2af] to [DELETE http://localhost:8100/session/54241DD8-4F5D-4158-AD13-CF305811F981] with no body
[FlutterDriver] Connection to ws://127.0.0.1:55130/8ecI9cXWD1M=/ws closed
[debug] [WD Proxy] Got response with status 200: {
[debug] [WD Proxy]   "value" : null,
[debug] [WD Proxy]   "sessionId" : "994547AE-82F4-487C-A0F4-9E0B547847A4"
[debug] [WD Proxy] }
[DevCon Factory] Releasing connections for 167319c03e0dee0e928f92be728b3a26433124e4 device on any port number
[DevCon Factory] Found cached connections to release: ["167319c03e0dee0e928f92be728b3a26433124e4:8100"]
[DevCon Factory] Releasing the listener for '167319c03e0dee0e928f92be728b3a26433124e4:8100'
[debug] [DevCon Factory] Cached connections count: 0
[debug] [XCUITest] Not clearing log files. Use `clearSystemFiles` capability to turn on.
[debug] [BaseDriver] Event 'quitSessionFinished' logged at 1580799274972 (12:24:34 GMT+0530 (India Standard Time))
[debug] [MJSONWP (060453e2)] Received response: null
[debug] [MJSONWP (060453e2)] But deleting session, so not returning
[debug] [MJSONWP (060453e2)] Responding to client with driver.deleteSession() result: null
[HTTP] <-- DELETE /wd/hub/session/060453e2-2368-4b53-9da9-d4424232699d 200 1045 ms - 76
[HTTP] 

java.lang.NoClassDefFoundError: kotlin/jvm/internal/Intrinsics

	at pro.truongsinh.appium_flutter.FlutterFinder.<init>(FlutterFinder.kt)
	at org.example.FlutterTest.setUp(FlutterTest.java:22)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: java.lang.ClassNotFoundException: kotlin.jvm.internal.Intrinsics
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 25 more`

**Comment by pradarttana-sharechat** [comment]
@ahashmi24 it is failing at 
`find = new FlutterFinder(driver);`

this is given as an example. I don't know what I am doing wrong.

**Comment by shibupanda** [comment]
@pradarttana-sharechat 
We are working on it to fix it permanently.
Time being please follow below.
https://github.com/truongsinh/appium-flutter-driver/issues/41

**Comment by ahashmi24** [comment]
> @ahashmi24
> It should be fixed from driver side but the time being please follow @vijaydeepak-rezi comment.
> please open different terminal and run iproxy 57172 57172 while running appium . It's port forwarding from local port to device port.

@shibupanda @truongsinh 
is it possible to prioritize this fix?
So that we can run the test on real iOS devices without any manual intervention.

**Comment by truongsinh** [comment]
@ahashmi24 there's a pending PR #62 to tackle this problem. Unfortunately I don't have a real iOS device myself thus can't ensure it works, can you try it yourself and give us feedback https://www.npmjs.com/package/appium-flutter-driver/v/0.0.24-1 ?
Please make sure that your system has `iproxy` command because this code relies on it.

**Comment by truongsinh** [comment]
@ahashmi24 any update?

**Comment by Hemanth-resideo** [comment]
@truongsinh  i have tried with the latest changes of flutter driver but the execution got stuck after it initiated iproxy command, it didn't
[Appium Logs.docx](https://github.com/truongsinh/appium-flutter-driver/files/4265406/Appium.Logs.docx)

 proceed further even after waiting for more than 5 mins.

Attached appium logs

**Comment by truongsinh** [comment]
Thanks for reporting, @Hemanth-Rezi, reading from the log, it seems it's doing fine until
```
iproxy 65105 65105 c5bf2cc84c87f60f2d4c2726b8e187086b05cfc5
```
The latest driver wait until this command returns, which I'm not sure it should (wait). Anybody here has experience with iproxy? Otherwise I really need to get my hands on a physical iphone before I can continue.

Anyway, it's likely that I can borrow an physical real iPhone on 10th of March.

**Comment by shibupanda** [comment]
Hi @truongsinh Even I don't have the iOS set up with me I may spend some time in my workspace.
I don't have much on iProxy but I may put some point which can help.

Please have a look on below link 
https://github.com/appium/appium-xcuitest-driver/blob/master/lib/device-connections-factory.js

If we can make a use of it I guess our problem will be resolved. Please let me know your thought on this.

**Comment by waitan2018** [comment]
## i also have this problem, after I use the iproxy command, i have new problems, hope you guys can help me, thank you very much:

[FlutterDriver] Attempt #1
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:55526/cNR0UNT9OdM=/ws
[FlutterDriver] Error: connect ECONNREFUSED 127.0.0.1:55526
[FlutterDriver]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1056:14)
[FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:55526/cNR0UNT9OdM=/ws
[FlutterDriver] Waiting 300 seconds before trying...
[FlutterDriver] Attempt #2
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:55526/cNR0UNT9OdM=/ws
[FlutterDriver] Connected to ws://127.0.0.1:55526/cNR0UNT9OdM=/ws
[FlutterDriver] Listing all isolates: [{"type":"@Isolate","id":"isolates/3792247070677151","name":"main","number":"3792247070677151"},{"type":"@Isolate","id":"isolates/2774744608590279","name":"main","number":"2774744608590279"}]
[FlutterDriver] "ext.flutter.driver" is not found in "extensionRPCs" ["ext.ui.window.scheduleFrame","ext.flutter.reassemble","ext.flutter.exit","ext.flutter.saveCompilationTrace","ext.flutter.platformOverride","ext.flutter.evict","ext.flutter.timeDilation","ext.flutter.debugPaint","ext.flutter.debugPaintBaselinesEnabled","ext.flutter.repaintRainbow","ext.flutter.debugCheckElevationsEnabled","ext.flutter.debugDumpLayerTree","ext.flutter.debugDumpRenderTree","ext.flutter.debugDumpSemanticsTreeInTraversalOrder","ext.flutter.debugDumpSemanticsTreeInInverseHitTestOrder","ext.flutter.debugDumpApp","ext.flutter.showPerformanceOverlay","ext.flutter.didSendFirstFrameEvent","ext.flutter.didSendFirstFrameRasterizedEvent","ext.flutter.profileWidgetBuilds","ext.flutter.debugAllowBanner","ext.flutter.debugWidgetInspector","ext.flutter.inspector.structuredErrors","ext.flutter.inspector.show","ext.flutter.inspector.trackRebuildDirtyWidgets","ext.flutter.inspector.trackRepaintWidgets","ext.flutter.inspector.disposeAllGroups","ext.flutter.inspector.disposeGroup","ext.flutter.inspector.isWidgetTreeReady","ext.flutter.inspector.disposeId","ext.flutter.inspector.setPubRootDirectories","ext.flutter.inspector.setSelectionById","ext.flutter.inspector.getParentChain","ext.flutter.inspector.getProperties","ext.flutter.inspector.getChildren","ext.flutter.inspector.getChildrenSummaryTree","ext.flutter.inspector.getChildrenDetailsSubtree","ext.flutter.inspector.getRootWidget","ext.flutter.inspector.getRootRenderObject","ext.flutter.inspector.getRootWidgetSummaryTree","ext.flutter.inspector.getDetailsSubtree","ext.flutter.inspector.getSelectedRenderObject","ext.flutter.inspector.getSelectedWidget","ext.flutter.inspector.getSelectedSummaryWidget","ext.flutter.inspector.isWidgetCreationTracked","ext.flutter.inspector.screenshot"]
[FlutterDriver] Waiting 300 seconds before trying...

**Comment by danielcYOU** [comment]
hi all, I was wondering if anyone managed to make any progress with this issue and get tests to run on real iOS Device? 
I have just tried the latest Appium Flutter Driver 0.0.24-1 but get the same issue mentioned in earlier comment of being stuck at the iproxy command.

[FlutterDriver] Running on iOS real device, doing "iproxy" now
[debug] [FlutterDriver] iproxy 60919 60919 cbfc3c66708111e5a48ad06f8917b951007bcb9e

for reference was using Appium 1.17.1 with iPhone 8 running OS 13.1.2

**Comment by KazuCocoa** [comment]
Appium stopped to use iproxy (https://github.com/libimobiledevice/libimobiledevice) to reduce 3rd party library issue since Appium 1.15.0. (the naming still remain though)
Instead, Appium communicates with `usbmuxd` directly via https://github.com/appium/appium-ios-device .
https://github.com/appium/appium-xcuitest-driver/blob/c7d8311559f53d8d7214ac0828e5bbf762762042/lib/driver.js#L492 may help the usage to bind the connection in xcuitest driver.

**Comment by alexshikov** [comment]
@waitan2018 I had the same issue and the fix was to add `enableFlutterDriverExtension();` to the `void main()` method of the application (not the test). So my `main.dart` looks like that:

```
import 'package:flutter_driver/driver_extension.dart';
// ... other imports

void main() {
  enableFlutterDriverExtension();

  WidgetsFlutterBinding.ensureInitialized();

  runApp(MyApp());
} 
```

**Comment by pripriya2** [comment]
Hi @KazuCocoa I am using Appium 1.16.0 and facing the Issue.
I could see it uses usbmuxd, but I am not able to understand that how would i find the workAround of this Issue
[FlutterDriver] Attempt #1
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:57496/zGG5BKi7gEI=/ws
[FlutterDriver] Error: connect ECONNREFUSED 127.0.0.1:57496
[FlutterDriver]     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16)
[FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:57496/zGG5BKi7gEI=/ws
[FlutterDriver] Waiting 300 seconds before trying.


As my flutterDriver is stuck.....................

**Comment by Hemanth-Rezi** [comment]
> Hi @KazuCocoa I am using Appium 1.16.0 and facing the Issue.
> I could see it uses usbmuxd, but I am not able to understand that how would i find the workAround of this Issue
> [FlutterDriver] Attempt #1
> [FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:57496/zGG5BKi7gEI=/ws
_**### > [FlutterDriver] Error: connect ECONNREFUSED 127.0.0.1:57496**_
> [FlutterDriver] at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16)
> [FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:57496/zGG5BKi7gEI=/ws
> [FlutterDriver] Waiting 300 seconds before trying.
> 
> As my flutterDriver is stuck.....................


@pripriya2 Based on your logs the port no ur device trying to connect  is 57496. Use command iproxy 57496 57496 to establish connection.
PS: every time when dart observatory is started the port num changes, u have to use the same port in which ur dart observatory is trying to connect

**Comment by ravijha-adi** [comment]
@Hemanth-Rezi  how to set the the static port for dart observatory? could you provide the step here please. And how about the Android with Windows do we need to set static port for adb forward tcp tcp?

**Comment by Hemanth-Rezi** [comment]
> @Hemanth-Rezi how to set the the static port for dart observatory? could you provide the step here please. And how about the Android with Windows do we need to set static port for adb forward tcp tcp?

@ravijha-adi 
For Mac machine just type **iproxy local port device port** in terminal
i haven't tried in windows machine but you can try with this command **adb forward tcp:6100 tcp:7100**

**Comment by ravijha-adi** [comment]
@Hemanth-Rezi  [debug] [FlutterDriver] C:\Program Files\Android\Sdk\platform-tools\adb.exe forward tcp:36988 tcp:36988 the tcp ip is coming at runtime and everytime it is different then how to handle this situation?

**Comment by shibupanda** [comment]
@ravijha-adi 
This issue is related to iOS (corner case) that we are working on it. Should not get the same for Android, mean no need to do port forwarding manually.
Could you please create a different issue for Android with all the details with the logs.

**Comment by shibupanda** [comment]
@truongsinh create a PR which include Spawn (child process).
Can we create a bump version on it.
https://github.com/truongsinh/appium-flutter-driver/pull/92

**Comment by nandhini0761** [comment]
Hi @KazuCocoa @shibupanda  I am using Appium 1.18.2 and facing the Issue.
I could see it uses usbmuxd, but I am not able to understand that how would i find the workAround of this Issue
[FlutterDriver] Attempt #1
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:57496/zGG5BKi7gEI=/ws
[FlutterDriver] Error: connect ECONNREFUSED 127.0.0.1:57496
[FlutterDriver] at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16)
[FlutterDriver] Check Dart Observatory URI ws://127.0.0.1:57496/zGG5BKi7gEI=/ws
[FlutterDriver] Waiting 300 seconds before trying.


Kindly help on this.

Advance Thanks

**Comment by ahmer-hash** [comment]
@shibupanda 
Do we have any capability to control this 300 seconds time for the second attempt?

[35m[FlutterDriver][39m Attempt #1
[35m[FlutterDriver][39m Connecting to Dart Observatory: ws://127.0.0.1:55774/fUECAwkOpxw=/ws
[35m[FlutterDriver][39m Error: connect ECONNREFUSED 127.0.0.1:55774
[35m[FlutterDriver][39m     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1144:16)
[35m[FlutterDriver][39m Check Dart Observatory URI ws://127.0.0.1:55774/fUECAwkOpxw=/ws
[35m[FlutterDriver][39m Waiting 300 seconds before trying...

I have tried to handle this manual intervention programmatically, to get Dart Observatory port [55774] from appium logs and hit it with iproxy command. However, it takes 5 minutes to start the execution on the real iPhone devices.

**Comment by shibupanda** [comment]
@ahmer-hash just know I publish a bump version to fix iOS real device issue. As I don't have real device could you please try the latest driver and share me the logs.
https://www.npmjs.com/package/appium-flutter-driver/v/0.0.26-1

**Comment by ahmer-hash** [comment]
@shibupanda 
appium-flutter-driver version ?
Currently, I am using below:
npm list -g appium-flutter-driver
/usr/local/lib
└─┬ appium@1.19.0-beta.0
  └── appium-flutter-driver@0.0.25

**Comment by shibupanda** [comment]
Yeah, but recently I publish latest bump version of driver for the fix. If you can check with latest driver that can help us to release proper release.

**Comment by ahmer-hash** [comment]
How can I get the latest driver? could you plz help me with the steps?

**Comment by shibupanda** [comment]
https://www.npmjs.com/package/appium-flutter-driver/v/0.0.26-1

Here you can get the latest flutter driver change the version in Appium package.json and do npm install.

**Comment by ahmer-hash** [comment]
@shibupanda 
It's failing after attempting 10 times. Please check logs [here](https://gist.github.com/ahmer-hash/db309b2976151d944c81f7678dc83878)

**Comment by shibupanda** [comment]
Looks like this time iProxy started. I need someone help to connect remotely to fix this issue as I don't have real iOS device. Could you please help me with your linked in account.

**Comment by anjanat11** [comment]
@shibupanda - This issue is a blocker for our testing also. As we need to test our scripts on real iOS device and not on simulator. Can you please let us know when the fix will be available?

**Comment by shibupanda** [comment]
Hi @anjanat11 
As I don't have an iOS set up was unable to test it and debug if someone can volunteer and connect with me than that would really help me to fix this quick.

**Comment by anjanat11** [comment]
@shibupanda - I can help out on the setup.

**Comment by shibupanda** [comment]
Thanks @anjanat11 please reach me in LinkedIn you can get it from my GitHub profile.

**Comment by anjanat11** [comment]
@shibupanda  - By using 0.0.26-1 driver the iproxy got connected automatically at the beginning itself but in the 1st attempt it failed because it is taking some time to initiate iproxy, but in second attempt it got connected and my script worked. In this version manual intervention is not required. 
Note -  I did not use retryBackoffTime capability.
Thanks for providing this driver. It was very helpful. Gr8 work.. 👍

**Comment by renjujv** [comment]
@shibupanda Thanks for your efforts, man.
@anjanat11 Thanks for letting us know. I'll also try this and report back.

**Comment by dvyAb** [comment]
Hello,

We have the same problem with flutter_driver, Have you found the solution ?, please

**Thanks**

**Comment by hex0cter** [comment]
@ajschwieterman Sorry for a silly question. I didn't see `iproxy` being called anywhere by default. Shall I rebuilt the library based on https://github.com/truongsinh/appium-flutter-driver/pull/62?

**Comment by ajschwieterman** [comment]
@hex0cter yes

**Comment by hex0cter** [comment]
ok I will give it a try. Thanks for the tip!

**Comment by shibupanda** [comment]
Hi All, The iOS related issue has been fixed in terms of iProxy.
Please use the latest appium-flutter-driver to make it work. 
[https://www.npmjs.com/package/appium-flutter-driver/v/0.0.28](url)

Thanks @KazuCocoa and @hex0cter for collaboration.

**Comment by ahashmi24** [comment]
@shibupanda 
I gave a try and have observed that it is working fine.

Thanks for the fix.

**Comment by ajschwieterman** [comment]
@ahashmi24
I'm still not able to launch Flutter apps on iOS (see #138).  I wonder if I am missing any dependencies, maybe **usbmuxd**?  Any help would be appreciated.

**Comment by pramod28m** [comment]
How to upgrade from v0.0.25 to 0.0.28 @shibupanda @KazuCocoa . 
I am seeing two Appium flutter driver version in laptop. By Default its taking v0.0.25 version when trying to launch the app in iOS real device.

**Comment by KazuCocoa** [comment]
Using `appium@beta` is the easiest way

**Comment by pramod28m** [comment]
Thanks @KazuCocoa .

**Comment by ravireddy07** [comment]
I'm getting this error, I've added below code in main.dart. But, still getting this error.

```
import 'package:flutter_driver/driver_extension.dart';
// ... other imports

void main() {
  enableFlutterDriverExtension();

  WidgetsFlutterBinding.ensureInitialized();

  runApp(MyApp());
} 
```

This is the ERROR:
```
`[FlutterDriver] Attempt #1
[FlutterDriver] Connecting to Dart Observatory: ws://127.0.0.1:34749/AY_kJYlMes8=/ws
[FlutterDriver] Connected to ws://127.0.0.1:34749/AY_kJYlMes8=/ws
[FlutterDriver] Listing all isolates: [{"type":"@Isolate","id":"isolates/3654605821806831","name":"main","number":"3654605821806831","isSystemIsolate":false}]
[FlutterDriver] "ext.flutter.driver" is not found in "extensionRPCs" ["ext.ui.window.scheduleFrame","ext.dart.io.getHttpEnableTimelineLogging","ext.dart.io.setHttpEnableTimelineLogging","ext.dart.io.getSocketProfile","ext.dart.io.startSocketProfiling","ext.dart.io.pauseSocketProfiling","ext.dart.io.clearSocketProfile","ext.dart.io.getVersion","ext.flutter.reassemble","ext.flutter.exit","ext.flutter.platformOverride","ext.flutter.brightnessOverride","ext.flutter.timeDilation","ext.flutter.evict","ext.flutter.debugPaint","ext.flutter.debugPaintBaselinesEnabled","ext.flutter.repaintRainbow","ext.flutter.debugCheckElevationsEnabled","ext.flutter.debugDumpLayerTree","ext.flutter.debugDumpRenderTree","ext.flutter.debugDumpSemanticsTreeInTraversalOrder","ext.flutter.debugDumpSemanticsTreeInInverseHitTestOrder","ext.flutter.debugDumpApp","ext.flutter.showPerformanceOverlay","ext.flutter.didSendFirstFrameEvent","ext.flutter.didSendFirstFrameRasterizedEvent","ext.flutter.fastReassemble","ext.flutter.profileWidgetBuilds","ext.flutter.invertOversizedImages","ext.flutter.debugAllowBanner","ext.flutter.debugWidgetInspector"]
[FlutterDriver] Waiting 3 seconds before trying...
`

Please help on this @alexshikov @waitan2018 if you know.

**Comment by KazuCocoa** [comment]
Have your app import https://github.com/appium-userland/appium-flutter-driver/blob/master/example/flutter_app_under_test/lib/main.dart#L2 flutter_driver package?
`"ext.flutter.driver" is not found in "extensionRPCs"` means the application under test does not have the `ext.flutter.driver` package. I guess you need to import them in your package like the link's example.

**Comment by just255** [comment]
I can't seem to get appium-flutter-driver working on iOS. Am I missing something? I tried the iproxy method, but also ran into an issue with that (address already in use), but assume it is not needed anymore with the recent changes. Any help is appreciated.
- I have appium-flutter-driver running fine on Android. 
- Flutter Driver extension is enabled with dependencies
- The app is up and running with an Observatory using ```fvm flutter run -d device_id --no-sound-null-safety```
```
An Observatory debugger and profiler on iPhone is available at: http://127.0.0.1:50476/PXwUHRx-Hig=/
```
- Appium version 
```
1.21.0
```
- Flutter Doctor
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.4 20F71 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
[✓] VS Code (version 1.54.1)
[✓] Connected device (2 available)

• No issues found!
```
- Desired Caps
```
capabilities.setCapability("noReset", "true");
capabilities.setCapability("autoLaunch", false);
capabilities.setCapability("platformName", "iOS");
capabilities.setCapability("platformVersion", "14.8");
capabilities.setCapability("deviceName", "iPhone 12");
capabilities.setCapability("automationName", "Flutter");
capabilities.setCapability("udid", "00008101-001A0CC00204001E");
capabilities.setCapability("bundleId", "com.tozelabs.tvshowtime.flutter");
capabilities.setCapability("xcodeSigningId", "iPhone Developer");
capabilities.setCapability("xcodeOrgId", "myOrgId");
```
- Appium log
```
[BaseDriver] Capability 'noReset' changed from string to boolean. This may cause unexpected behavior
[XCUITest] 'platformVersion' capability ('undefined') is not a valid version number. Consider fixing it or be ready to experience an inconsistent driver behavior.
[BaseDriver] Session created with session id: 426ad864-0a0b-4187-8800-6202dc1b6125
[debug] [XCUITest] Current user: 'jsimmons'
[debug] [XCUITest] Available devices: 00008101-001A0CC00204001E
[debug] [XCUITest] Creating iDevice object with udid '00008101-001A0CC00204001E'
[XCUITest] Determining device to run tests on: udid: '00008101-001A0CC00204001E', real device: true
[XCUITest] No platformVersion specified. Using device version: '14.8.1'
[XCUITest] Normalized platformVersion capability value '14.8.1' to '14.8'
[debug] [BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1637760455909 (05:27:35 GMT-0800 (Pacific Standard Time))
[debug] [BaseDriver] Event 'appConfigured' logged at 1637760455910 (05:27:35 GMT-0800 (Pacific Standard Time))
[debug] [BaseDriver] Event 'resetStarted' logged at 1637760455911 (05:27:35 GMT-0800 (Pacific Standard Time))
[debug] [XCUITest] Reset: fullReset not set. Leaving as is
[debug] [BaseDriver] Event 'resetComplete' logged at 1637760455911 (05:27:35 GMT-0800 (Pacific Standard Time))
[WebDriverAgent] Using WDA path: '/usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent'
[WebDriverAgent] Using WDA agent: '/usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj'
[debug] [XCUITest] Crash reports root '/Users/jsimmons/Library/Logs/CrashReporter/MobileDevice/Justin’s iPhone' does not exist. Got nothing to gather.
[debug] [BaseDriver] Event 'logCaptureStarted' logged at 1637760456151 (05:27:36 GMT-0800 (Pacific Standard Time))
[XCUITest] Setting up real device
[debug] [WebDriverAgent] No obsolete cached processes from previous WDA sessions listening on port 8100 have been found
[DevCon Factory] Requesting connection for device 00008101-001A0CC00204001E on local port 8100, device port 8100
[debug] [DevCon Factory] Cached connections count: 1
[DevCon Factory] Found cached connections on port #8100: ["00008101-001A0CC00204001E:8100"]
[DevCon Factory] Port #8100 is busy. Did you quit the previous driver session(s) properly?
[DevCon Factory] Trying to release the port
[DevCon Factory] Releasing the listener for '00008101-001A0CC00204001E:8100'
[debug] [iProxy@00008101:8100] Closing the connection
[iProxy@00008101:8100] The connection has been closed
[DevCon Factory] Port #8100 has been successfully released after 2ms
[DevCon Factory] Successfully requested the connection for 00008101-001A0CC00204001E:8100
[debug] [XCUITest] Starting WebDriverAgent initialization with the synchronization key 'XCUITestDriver'
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[debug] [WD Proxy] Got response with status 200: {"value":{"message":"WebDriverAgent is ready to accept commands","state":"success","os":{"testmanagerdVersion":28,"name":"iOS","sdkVersion":"15.0","version":"14.8.1"},"ios":{"ip":"169.254.125.167"},"ready":true,"build":{"upgradedAt":"499162500000","time":"Nov 23 2021 17:19:57","productBundleIdentifier":"com.facebook.WebDriverAgentRunner"}},"sessionId":"20896A9F-2C16-43C7-996E-AD0B838CCE90"}
[debug] [WebDriverAgent] Upgrade timestamp of the currently bundled WDA: 499162500000
[debug] [WebDriverAgent] Upgrade timestamp of the WDA on the device: 499162500000
[WebDriverAgent] Will reuse previously cached WDA instance at 'http://127.0.0.1:8100/' with 'com.facebook.WebDriverAgentRunner'. Set the wdaLocalPort capability to a value different from 8100 if this is an undesired behavior.
[debug] [XCUITest] Trying to start WebDriverAgent 1 times with 10000ms interval
[debug] [XCUITest] These values can be customized by changing wdaStartupRetries/wdaStartupRetryInterval capabilities
[debug] [BaseDriver] Event 'wdaStartAttempted' logged at 1637760456567 (05:27:36 GMT-0800 (Pacific Standard Time))
[WebDriverAgent] Using provided WebdriverAgent at 'http://127.0.0.1:8100/'
[debug] [WD Proxy] Matched '/status' to command name 'getStatus'
[debug] [WD Proxy] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
[debug] [WD Proxy] Got response with status 200: {"value":{"message":"WebDriverAgent is ready to accept commands","state":"success","os":{"testmanagerdVersion":28,"name":"iOS","sdkVersion":"15.0","version":"14.8.1"},"ios":{"ip":"169.254.125.167"},"ready":true,"build":{"upgradedAt":"499162500000","time":"Nov 23 2021 17:19:57","productBundleIdentifier":"com.facebook.WebDriverAgentRunner"}},"sessionId":"20896A9F-2C16-43C7-996E-AD0B838CCE90"}
[debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1637760456580 (05:27:36 GMT-0800 (Pacific Standard Time))
[debug] [XCUITest] Sending createSession command to WDA
[debug] [WD Proxy] Matched '/session' to command name 'createSession'
[debug] [WD Proxy] Proxying [POST /session] to [POST http://127.0.0.1:8100/session] with body: {"capabilities":{"firstMatch":[{"arguments":[],"environment":{},"eventloopIdleDelaySec":0,"shouldWaitForQuiescence":true,"shouldUseTestManagerForVisibilityDetection":false,"maxTypingFrequency":60,"shouldUseSingletonTestManager":true,"shouldTerminateApp":true}],"alwaysMatch":{}}}
[debug] [WD Proxy] Got response with status 200: {"value":{"sessionId":"66B285B2-58A6-4275-B9E0-8D3B875E4C35","capabilities":{"device":"iphone","browserName":"TV Time Flutter","sdkVersion":"14.8.1","CFBundleIdentifier":"com.tozelabs.tvshowtime.flutter"}},"sessionId":"66B285B2-58A6-4275-B9E0-8D3B875E4C35"}
[WD Proxy] Determined the downstream protocol as 'W3C'
[debug] [BaseDriver] Event 'wdaSessionStarted' logged at 1637760456859 (05:27:36 GMT-0800 (Pacific Standard Time))
[debug] [BaseDriver] Event 'wdaStarted' logged at 1637760456859 (05:27:36 GMT-0800 (Pacific Standard Time))
[XCUITest] Skipping setting of the initial display orientation. Set the "orientation" capability to either "LANDSCAPE" or "PORTRAIT", if this is an undesired behavior.
[debug] [BaseDriver] Event 'orientationSet' logged at 1637760456859 (05:27:36 GMT-0800 (Pacific Standard Time))
[debug] [BaseDriver] The value of 'elementResponseAttributes' setting did not change. Skipping the update for it
[debug] [BaseDriver] The value of 'shouldUseCompactResponses' setting did not change. Skipping the update for it
[debug] [FlutterDriver] Deleting Flutter Driver session
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1637760456862 (05:27:36 GMT-0800 (Pacific Standard Time))
[debug] [W3C] Encountered internal error running command: TypeError: Cannot read property 'match' of undefined
[debug] [W3C]     at Object.exports.processLogToGetobservatory (/usr/local/lib/node_modules/appium/node_modules/appium-flutter-driver/build/driver/lib/sessions/observatory.js:118:9)
[debug] [W3C]     at Object.exports.getObservatoryWsUri (/usr/local/lib/node_modules/appium/node_modules/appium-flutter-driver/lib/sessions/ios.ts:31:21)
[debug] [W3C]     at Object.exports.startIOSSession (/usr/local/lib/node_modules/appium/node_modules/appium-flutter-driver/lib/sessions/ios.ts:23:34)
[debug] [W3C]     at runMicrotasks (<anonymous>)
[debug] [W3C]     at processTicksAndRejections (node:internal/process/task_queues:94:5)
[debug] [W3C]     at FlutterDriver.exports.createSession (/usr/local/lib/node_modules/appium/node_modules/appium-flutter-driver/lib/sessions/session.ts:15:47)
[debug] [W3C]     at AppiumDriver.createSession (/usr/local/lib/node_modules/appium/lib/appium.js:387:35)
[HTTP] <-- POST /wd/hub/session 500 1197 ms - 579
[HTTP] 
[debug] [WebDriverAgent] Parsed BUILD_DIR configuration value: '/Users/jsimmons/Library/Developer/Xcode/DerivedData/WebDriverAgent-ciegwgvxzxdrqthilmrmczmqvrgu/Build/Products'
[debug] [WebDriverAgent] Got derived data root: '/Users/jsimmons/Library/Developer/Xcode/DerivedData/WebDriverAgent-ciegwgvxzxdrqthilmrmczmqvrgu'
```

**Comment by KazuCocoa** [comment]
What about the latest appium? 1.21.0 had flutter driver 0.0.29 while 1.22.0 has 0.0.31.
A newer one has an adjustment to check the websocket port for flutter.

**Comment by just255** [comment]
> What about the latest appium? 1.21.0 had flutter driver 0.0.29 while 1.22.0 has 0.0.31. A newer one has an adjustment to check the websocket port for flutter.

I tried that one, but failed to get WDA running with it. I will give 1.22 another go and see what happens

**Comment by just255** [comment]
I managed to get WDA running and the issue still persists with using Appium 1.22.1. Just to clarify my environment a bit more:
1. Running on MacOS Big Sur
2. Xcode 13
3. App is signed using a non enterprise cert (dev cert only, so I can't reference the Runner.app without it constantly crashing), which prevents me from referencing the app in desired caps like I do with Android. Unless there's another way I'm not aware of.

And just so I know I'm performing the steps correctly, I should:
1. Start appium
2. Do a ```flutter run``` to install and run the app and kick off the Observer
3. Create the Flutter session

**Comment by KazuCocoa** [comment]
Afaik, appium-flutter-driver tries to find a websocket url when the AUT starts. 
Theoretically, if the websocket url in the system log (iOS, android is logcat) appeared before starting appium-flutter-driver's URL observatory method, the driver probably cannot find the URL. It results an error.

With 1.22.0, what log could you see?
Like https://github.com/appium-userland/appium-flutter-driver/blob/16235176c2c8b75e87b4b63c898b553963c10593/driver/lib/sessions/observatory.ts#L18 waiting log loop appeared on the log and got an error, probably the flutter driver was not able to find the URL by the above reason.

**Comment by just255** [comment]
It fails with the same issue as stated above. Looking at the line that specifies the Observatory, I see that it does not match https://github.com/appium-userland/appium-flutter-driver/blob/20ac4a95b78f19e61765fc39823b377609fe4ae5/driver/lib/sessions/observatory.ts#L143 
Instead it says ```An Observatory debugger and profiler on Justin’s iPhone is available at: http://127.0.0.1:52817/_w_SwaKs9-g=/```. Could this be because I'm using Flutter 2.5.2? Either that or I'm missing something when running the application.

**Comment by KazuCocoa** [comment]
https://github.com/flutter/flutter/blob/f90b019c68edf4541a4c8273865a2b40c2c01eb3/dev/devicelab/lib/framework/runner.dart#L183

https://github.com/flutter/flutter/blob/52ae102f182afaa0524d0d01d21b2d86d15a11dc/packages/flutter_tools/lib/src/resident_runner.dart#L1386-L1389

oh, good to know. Yea, `// Caution: This log line is parsed by device lab tests.` the line addressed so.

**Comment by just255** [comment]
I tried the above locally and it's still not finding the Observatory. I added a debug line before the ```match``` check ```  logger_1.log.debug(adbLogs.map((e) => e.message));``` and the Observatory is nowhere in there. What logs are being parsed? I'm running the ```flutter run``` command to start the app in terminal (I can't launch the app with the driver), so that's where I'm seeing my Observatory. Would it be possible to write my output from terminal to whatever log is being parsed?

**Comment by KazuCocoa** [comment]
I guess this driver did not expect to launch the app under test outside this driver.

I wonder the below steps:

1. Start an appium session without `app` and `appPackage` (android)
2. runs `flutter run` while the 1) calling https://github.com/appium-userland/appium-flutter-driver/blob/master/driver/lib/sessions/observatory.ts#L15

basically the observatory watches logcat in Android, system logs in iOS. The observatory url must be in lohgcat/system log while the observatory duration. (the above link)
If your `flutter run` needs more time, then you can increase the value of `maxRetryCount` and `retryBackoffTime`. Then, the watching the observatory url time may increase.

**Comment by just255** [comment]
That also fails. The problem I think is that the driver is missing the context of my terminal output. One solution might be to add a capability to supply the Observatory manually.

**Comment by KazuCocoa** [comment]
> One solution might be to add a capability to supply the Observatory manually.

Yes, it could be a workaround. So, you can add a new capability like `observatoryWs` to give websocket URL.
https://github.com/appium-userland/appium-flutter-driver/blob/c0da23edcd8dfe10bc5ae4d59ba04d33da7ad0ff/driver/lib/sessions/android.ts#L39
https://github.com/appium-userland/appium-flutter-driver/blob/c0da23edcd8dfe10bc5ae4d59ba04d33da7ad0ff/driver/lib/sessions/ios.ts#L77

**Comment by jeanpier0425** [comment]
> @waitan2018 I had the same issue and the fix was to add `enableFlutterDriverExtension();` to the `void main()` method of the application (not the test). So my `main.dart` looks like that:
> 
> ```
> import 'package:flutter_driver/driver_extension.dart';
> // ... other imports
> 
> void main() {
>   enableFlutterDriverExtension();
> 
>   WidgetsFlutterBinding.ensureInitialized();
> 
>   runApp(MyApp());
> } 
> ```
@waitan2018
@KazuCocoa 
@renjujv 
I did exactly the same, but my problem is that it still takes 300 seconds to interact with the app.
Is there any way to reduce that time?

**Comment by KazuCocoa** [comment]
I guess it is in Dart VM stuff. then, I guess only `wait` something in https://github.com/appium-userland/appium-flutter-driver is what appium-flutter-driver can do...
I'm not familiar with Flutter/Dart implementation, so do not have any ideas about the layer.

**Comment by wswebcreation** [comment]
I had the same issue and wanted to share what solved it for me. When I used 

`flutter build ipa --export-method development`

I got the error

```log
6:46:31 V [debug] [AppiumDriver@41d9] Encountered internal error running command: Error: No observatory url was found in the device log. Please make sure the application under test logs an observatory url when the application starts.
```

But when I used 

`flutter build ipa --profile`

it worked like a charm.

**Comment by LuizHenrique78** [comment]
Hi guys, I`m still haing this issue while testing a flutter app on a real iOS device. I`ve tried to start the iproxy but wint worked.

**Comment by rakhmatullov** [comment]
Same issue for me Flutter app with iOS 17

**Comment by baselalali** [comment]
Same Problem with: 
`[Appium]   - xcuitest@4.35.0 (automationName 'XCUITest')
[Appium]   - uiautomator2@2.29.8 (automationName 'UiAutomator2')
[Appium]   - flutter@1.22.0 (automationName 'Flutter')
`
iOS 16

**Comment by rakhmatullov** [comment]
From our side the problem was on the app side configuration and "flutter build" command. We've resolved that, all works fine.
